plugin/src/main/java/net/dzikoysk/funnyguilds/user/User.java

```diff
@@ -201,9 +149,44 @@ public BossBarProvider getBossBar() {
         return this.bossBarProvider;
     }
 
-    @Override
-    public EntityType getType() {
-        return EntityType.USER;
+    public UserProfile getProfile() {
+        return this.profile;
+    }
+
+    public void setProfile(UserProfile profile) {
+        this.profile = profile;
+    }
+
+    // Deprecated methods
+    //  - use UserGameProfile
+    //  - add methods to UserGameProfile and use them
+    //  - get Player/OfflinePlayer in the another way
+
+    @Deprecated
+    public OfflinePlayer getOfflinePlayer() {
+        return Bukkit.getOfflinePlayer(this.uuid);
+    }
+
+    @Deprecated
+    public Option<Player> getPlayer() {
+        Player player = this.playerRef.get();
+        if (player != null) {
+            return Option.of(player);
+        }
+
+        player = Bukkit.getPlayer(this.uuid);
+        if (player != null) {
+            this.playerRef = new WeakReference<>(player);
+            return Option.of(player);
+        }
+
+        return Option.none();
+    }
+
+    @Deprecated
+    public void updateReference(Player player) {
```

Comment:
Remove this method and playerReference field (use UserProfile everywhere)

Reply:
for future removal